src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -2012,13 +2020,24 @@ class TerminalInstanceDragAndDropController extends Disposable implements IDragA
 		this._onDropFile.fire(path);
 	}
 
-	private _getDropSide(e: DragEvent): 'left' | 'right' {
+	private _getDropSide(e: DragEvent): 'before' | 'after' {
 		const target = this._container;
 		if (!target) {
-			return 'right';
+			return 'after';
 		}
+
 		const rect = target.getBoundingClientRect();
-		return e.clientX - rect.left < rect.width / 2 ? 'left' : 'right';
+		return this._getViewOrientation() === Orientation.HORIZONTAL
+			? (e.clientX - rect.left < rect.width / 2 ? 'before' : 'after')
+			: (e.clientY - rect.top < rect.height / 2 ? 'before' : 'after');
+	}
+
+	private _getViewOrientation(): Orientation {
+		const panelPosition = this._layoutService.getPanelPosition();
+		const terminalLocation = this._viewDescriptorService.getViewLocationById(TERMINAL_VIEW_ID);
+		return terminalLocation === ViewContainerLocation.Panel && panelPosition === Position.BOTTOM
+			? Orientation.HORIZONTAL
+			: Orientation.VERTICAL;
```

Comment:
Had to copy this logic from `TerminalGroup` as I cannot use `TerminalService.onPanelOrientationChanged` to avoid a cyclic dependency